pkg/logs/internal/util/testutils/hostname_mock.go

```diff
@@ -0,0 +1,41 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package testutils
+
+import (
+	"context"
+
+	"github.com/DataDog/datadog-agent/comp/core/hostname/hostnameinterface"
+)
+
+type mockHostnameService struct {
+	name string
+}
+
+func (m *mockHostnameService) Get(_ context.Context) (string, error) {
```

Comment:
[We already have a hostname mock ](https://github.com/DataDog/datadog-agent/blob/ff99122acb83f841a058d4ed7ba59f76e73a862e/comp/core/hostname/hostnameimpl/service_mock.go#L36) - can we re-use this instead of duplicating all the code?